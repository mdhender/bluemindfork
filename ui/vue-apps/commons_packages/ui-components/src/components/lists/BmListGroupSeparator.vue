<script>
export default {
    name: "BmListGroupSeparator",
    props: {
        tag: {
            type: String,
            default: "div"
        }
    },
    render(h) {
        return h(this.$props.tag, { class: "list-group-separator" }, this.$slots.default);
    }
};
</script>

<style lang="scss">
@import "../../css/utils/variables.scss";

.list-group-separator {
    position: relative;
    display: block;
    padding: $sp-2;
}
</style>
