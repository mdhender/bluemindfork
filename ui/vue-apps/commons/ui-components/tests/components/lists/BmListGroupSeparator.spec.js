import { mount } from "@vue/test-utils";
import BmListGroupSeparator from "../../../src/components/lists/BmListGroupSeparator";

describe("BmListGroupSeparator", () => {
    function defaultMount() {
        return mount(BmListGroupSeparator, {
            propsData: {
                tag: "span"
            }
        });
    }

    test("is a Vue instance", () => {
        expect(defaultMount().vm).toBeTruthy();
    });

    test("BmListGroupSeparator should match snapshot", () => {
        expect(defaultMount().vm.$el).toMatchSnapshot();
    });

    test("BmListGroupSeparator have a list-group-separator class", () => {
        expect(defaultMount().vm.$el.className).toEqual(expect.stringContaining("list-group-separator"));
    });

    test("BmListGroupSeparator html tag match property", () => {
        expect(defaultMount().vm.$el.tagName).toEqual("SPAN");
    });
});
