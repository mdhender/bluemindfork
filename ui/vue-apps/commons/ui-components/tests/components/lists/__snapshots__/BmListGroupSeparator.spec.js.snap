// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BmListGroupSeparator BmListGroupSeparator should match snapshot 1`] = `
<span
  class="list-group-separator"
/>
`;
